### PR TITLE
Testing MOB-4125

### PR DESCRIPTION
**Pull request by mateid-kr**
## JIRA
[MOB-4125](https://karhoo.atlassian.net/browse/MOB-4125)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [X] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [X] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: app/src/main/java/com/karhoo/samples/uisdk/dropin/MainActivity.kt [hunk 1]
```diff
@@ -154,9 +159,30 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
+    var requestedAuthentication = false
     private fun applyAdyenTokenExchangeConfig() {
         val config = AdyenTokenExchangeConfig(applicationContext)
         config.paymentManager = adyenPaymentManager
+        config.externalAuthenticationNeeded = {
+            if (!requestedAuthentication) {
+                Log.e("matei", "Need an external authentication")
```
Comment (dianakarh) [inline app/src/main/java/com/karhoo/samples/uisdk/dropin/MainActivity.kt:168]: I see you signed your work @mateid-kr :))

File: app/src/main/java/com/karhoo/samples/uisdk/dropin/config/KarhooConfig.kt [hunk 2]
```diff
@@ -43,4 +43,8 @@ class KarhooConfig(val context: Context, private val authMethod: AuthenticationM
     override fun analyticsProvider(): AnalyticProvider? {
         return null
     }
+
+    override suspend fun requireSDKAuthentication(callback: () -> Unit) {
+
```
Comment (dianakarh) [inline app/src/main/java/com/karhoo/samples/uisdk/dropin/config/KarhooConfig.kt:48]: Shouldn't all these have implementations in order for the client app to work properly when testing over longer periods of time?

File: app/src/main/java/com/karhoo/samples/uisdk/dropin/MainActivity.kt (hunk 1 above)
Comment (mateid-kr) [reply app/src/main/java/com/karhoo/samples/uisdk/dropin/MainActivity.kt:168]: It's still in development 😄 . The final PR won't contain these

**Review by mionescu95 — APPROVED** [review]

File: uisdk/README.md [hunk 3]
```diff
@@ -65,6 +65,17 @@ class KarhooConfig(val context: Context): KarhooUISDKConfiguration {
     override fun authenticationMethod(): AuthenticationMethod {
         return AuthenticationMethod.KarhooUser()
     }
+
+    override suspend fun requireSDKAuthentication(callback: () -> Unit) {
+       KarhooApi.userService.logout()
+       KarhooApi.authService.login(token).execute { result ->
+            when (result) {
+                is Resource.Success -> callback.invoke()
+                is Resource.Failure -> //Show error
```
Comment (dianakarh) [inline uisdk/README.md:74]: Shouldn't we show an error here?

**Review by dianakarh — APPROVED** [review]

**Pull request merged**